src/main/java/frc/trigon/robot/subsystems/swerve/SwerveModule.java

```diff
@@ -56,11 +73,32 @@ void setDriveMotorClosedLoop(boolean closedLoop) {
     }
 
     void setTargetState(SwerveModuleState targetState) {
-        this.targetState = SwerveModuleState.optimize(targetState, getCurrentAngle());
+        targetState.optimize(getCurrentAngle());
+        this.targetState = targetState;
         setTargetAngle(this.targetState.angle);
         setTargetVelocity(this.targetState.speedMetersPerSecond, this.targetState.angle);
     }
 
+    void setTargetState(SwerveModuleState swerveModuleState, double targetCurrent) {
```

Comment:
jdoc 
explain when this should be used, and the diff between the other function